### PR TITLE
+ Option.ofPair

### PR DESCRIPTION
**Pull request by 3Rafal**
Add function for conversion of C# try pattern to F# option.

### CODE REVIEW

**Comment by 3Rafal** [comment]
This is an implementation of #408 idea

**Comment by wallymathieu** [comment]
Looks like we get:
`System.Xml.XmlException: Name cannot begin with the ''' character, hexadecimal value 0x27. Line 6, position 53` as an error.

File: tests/FSharpPlus.Tests/Extensions.fs [hunk 1]
```diff
@@ -342,6 +342,16 @@ module Extensions =
     Nullable.toList (Nullable 1) |> areEqual [1]
     Nullable.toList (Nullable()) |> areEqual []
  
+  [<Test>]
+  let ``Option.fromPair returns Some when operation succeeds`` () =
```
Comment (gusty) [inline tests/FSharpPlus.Tests/Extensions.fs:346]: It's `ofPair` not `fromPair` ;)

Comment (3Rafal) [reply tests/FSharpPlus.Tests/Extensions.fs]: Changed ;)

**Comment by 3Rafal** [comment]
> 
> 
> Looks like we get:
> `System.Xml.XmlException: Name cannot begin with the ''' character, hexadecimal value 0x27. Line 6, position 53` as an error.
> from the XML-documentation.

Fixed :-)

**Review by gusty — COMMENTED** [review]
This looks OK.

Now the question is does the name of the function makes sense?
Do we all agree?
Or are there better proposals?

@wallymathieu @adz @cannorin @cmeeren

**Comment by wallymathieu** [comment]
I can't come up with a better proposal.

**Comment by cmeeren** [comment]
@gusty Thanks for thinking of me, but I no longer use FSharpPlus due to the IDE performance issues.

**Comment by gusty** [comment]
@cmeeren  Sad to hear that after having you collaborating actively by reporting issues and requesting features for some years.

This function is for Extensions. Does the extension module degrade IDE as well?

Or maybe you don't find Extensions that useful for your projects?

**Comment by cmeeren** [comment]
I haven't considered using only Extensions. Will keep it in mind if I encounter pain points, but I'm doing fine for now. :)

**Comment by gusty** [comment]
Maybe I should find some time again to improve the F# compiler speed for constraint solving.

Thanks one more time for your feedback.

Hopefully is not the last one ;)

**Comment by wallymathieu** [comment]
I think the best bet here is to go with `Option.ofPair` since we don't have any ideas about any better naming.

**Pull request merged**